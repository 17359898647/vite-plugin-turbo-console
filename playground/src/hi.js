export function hi() {
  const helloJS = "hello JS"
  console.log(helloJS)
}
