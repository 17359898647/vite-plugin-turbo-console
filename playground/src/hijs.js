export function hijs() {
  const helloJS = "hello JS"
  console.log(helloJS)
}
