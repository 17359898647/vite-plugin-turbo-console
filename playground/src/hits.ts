export function hits() {
  const helloTS = "hello TS"
  console.log(helloTS)
}
