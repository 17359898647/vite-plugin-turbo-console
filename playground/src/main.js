import './assets/main.css'

import { createApp } from 'vue'
import App from './App.vue'

const currentFile = 'main.js'
// console.log("ðŸš€ ~ file: main.js:7 ~ currentFile:", currentFile)
console.log(currentFile)

createApp(App).mount('#app')
