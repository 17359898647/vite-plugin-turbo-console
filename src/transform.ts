import { basename, extname } from 'node:path'

const transformFileTypes = ['.js', '.jsx', '.ts', '.tsx', '.vue', '.svelte', '.astro']

export function transformCode(code: string, path: string) {
  let newCode = ''
  if (transformFileTypes.includes(extname(path)) && !path.includes('node_modules')) {
    const lines = code.split('\n')
    for (let i = 0; i < lines.length; i++) {
      if (/^\s*console\.log\([^\)]*\)\s*;?\s*$/.test(lines[i])) {
        const consoleContent = lines[i].match(/s*console\.log\((.+)\)/)
        if (consoleContent)
          newCode += `console.log('%c${basename(path)}:${i + 1} ~ ${consoleContent[1]}:', '${consoleStyle(extname(path))}' ,${consoleContent[1]})\n`
        else
          newCode += `${lines[i]}\n`
      }

      else {
        newCode += `${lines[i]}\n`
      }
    }
  }
  else {
    newCode = code
  }

  return newCode
}

function consoleStyle(fileType: string): string {
  let styleString = 'padding:4px; border-radius:5px; font-weight:600; '
  switch (fileType) {
    case '.js':
    case '.jsx':
      styleString += 'color: #111827; background:#F7DF1E'
      break
    case '.ts':
    case '.tsx':
      styleString += 'color: #fff; background:#3178C6'
      break
    case '.vue':
      styleString += 'color: #fff; background:#4FC08D'
      break
    case '.svelte':
      styleString += 'color: #fff; background:#FF3E00'
      break
    case '.astro':
      styleString += 'color: #fff; background:#FF5D01'
      break
    default:
      styleString += 'color: #111827; background:#F7DF1E'
      break
  }

  return styleString
}
