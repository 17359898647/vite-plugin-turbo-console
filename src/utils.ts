export const transformFileTypes = ['.js', '.jsx', '.ts', '.tsx', '.vue', '.svelte', '.astro']

const commonStyle = 'padding:4px; border-radius:5px; font-weight:600; '

const consoleStyles: Record<string, string> = {
  '.js': `${commonStyle}color: #111827; background: #F7DF1E`,
  '.jsx': `${commonStyle}color: #111827; background: #F7DF1E`,
  '.ts': `${commonStyle}color: #fff; background: #3178C6`,
  '.tsx': `${commonStyle}color: #fff; background: #3178C6`,
  '.vue': `${commonStyle}color: #fff; background: #4FC08D`,
  '.svelte': `${commonStyle}color: #fff; background: #FF3E00`,
  '.astro': `${commonStyle}color: #fff; background: #FF5D01`,
  'default': `${commonStyle}color: #111827; background: #F7DF1E`,
}

export function getConsoleStyle(fileType: string): string {
  return consoleStyles[fileType] ?? consoleStyles.default
}
