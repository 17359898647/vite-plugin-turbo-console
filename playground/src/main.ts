import './assets/main.css'

import { createApp } from 'vue'
import App from './App.vue'
import {hi} from './hi.js'

const helloTS = "hello TS"
console.log(helloTS)
hi()

createApp(App).mount('#app')
