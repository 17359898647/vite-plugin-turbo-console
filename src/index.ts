import type { PluginOption } from 'vite'
import { transformCode } from './transform'

function VitePluginTurboConsole(): PluginOption {
  return {
    name: 'vite-plugin-turbo-console',
    enforce: 'pre',
    apply: 'serve',
    transform(code, id) {
      return transformCode(code, id)
    },
  }
}

export default VitePluginTurboConsole
export type { PluginOption }
