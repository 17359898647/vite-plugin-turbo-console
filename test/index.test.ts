import { describe, expect, test } from 'vitest'
import { transformCode } from '../src/transform'

const jsTest = {
  code: 'console.log(foo,bar)',
  path: '/path/to/main.js',
}

const tsTest = {
  code: 'console.log(foo,bar)',
  path: '/path/to/main.ts',
}

const vueTest = {
  code: 'console.log(foo,bar)',
  path: '/path/to/main.vue',
}

describe('transformCode', () => {
  test('jsCode', () => {
    expect(transformCode(jsTest.code, jsTest.path)).toMatchSnapshot()
  })

  test('tsCode', () => {
    expect(transformCode(tsTest.code, tsTest.path)).toMatchSnapshot()
  })

  test('vueCode', () => {
    expect(transformCode(vueTest.code, vueTest.path)).toMatchSnapshot()
  })
})
