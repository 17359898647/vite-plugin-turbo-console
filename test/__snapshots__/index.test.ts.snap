// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`transformCode > jsCode 1`] = `
"console.log('%cmain.js:1 ~ foo,bar:', 'padding:4px; border-radius:5px; font-weight:600; color: #111827; background:#F7DF1E' ,foo,bar)
"
`;

exports[`transformCode > tsCode 1`] = `
"console.log('%cmain.ts:1 ~ foo,bar:', 'padding:4px; border-radius:5px; font-weight:600; color: #fff; background:#3178C6' ,foo,bar)
"
`;

exports[`transformCode > vueCode 1`] = `
"console.log('%cmain.vue:1 ~ foo,bar:', 'padding:4px; border-radius:5px; font-weight:600; color: #fff; background:#4FC08D' ,foo,bar)
"
`;
